File: tests/fixtures/idioms/go/duplicate_test.go
package tests

func HandlerA(req string) error {
    validate(req)
    save(req)
    notify(req)
    return nil
}

func HandlerB(req string) error {
    validate(req)
    save(req)
    notify(req)
    return nil
}

func HandlerC(req string) error {
    validate(req)
    save(req)
    notify(req)
    return nil
}

File: tests/fixtures/idioms/python/duplicate_test.py
"""Same control-flow copied to many tests — true positive finding, should be reported."""
def test_path_a():
    validate("a")
    save("a")
    notify("a")

def test_path_b():
    validate("b")
    save("b")
    notify("b")

def test_path_c():
    validate("c")
    save("c")
    notify("c")
